packages/backend/src/services/CatalogService/CatalogService.ts

```diff
@@ -346,26 +343,23 @@ export class CatalogService<
     ) => {
         // TODO move to space utils ?
         const spaces = await this.spaceModel.find({ projectUuid });
+        const spacesAccess = await this.spaceModel.getUserSpacesAccess(
+            user.userUuid,
+            spaces.map((s) => s.uuid),
+        );
 
-        const hasSpaceAccess = await Promise.all(
-            spaces.map(async (space) =>
+        const allowedSpaceUuids = spaces
+            .filter((space) =>
                 user.ability.can(
                     'view',
                     subject('Space', {
                         organizationUuid: space.organizationUuid,
                         projectUuid,
                         isPrivate: space.isPrivate,
-                        access: await this.spaceModel.getUserSpaceAccess(
-                            user.userUuid,
-                            space.uuid,
-                        ),
+                        access: spacesAccess[space.uuid] ?? [],
```

Comment:
what happens if I forget to add `?? []` as a lazy dev?

Reply:
We need to enable typescript `--noUncheckedIndexedAccess` for it to throw an error since atm typescript assumes `spacesAccess[space.uuid]` is always defined. 

https://github.com/Microsoft/TypeScript/issues/13778